src/util/psbt/mod.rs

```diff
@@ -36,15 +36,53 @@ use crate::util::bip32::{ExtendedPubKey, KeySource};
 /// Partially signed transaction, commonly referred to as a PSBT.
 pub type Psbt = PartiallySignedTransaction;
 
+/// Future version of PSBT this library can't parse
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct FutureVersionError(u32);
+
+impl fmt::Display for FutureVersionError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("Unknown future PSBT version ")?;
+        fmt::Display::fmt(&self.0, f)
+    }
+}
+
+#[cfg(feature = "std")]
+impl ::std::error::Error for FutureVersionError {}
+
+/// Partially signed transaction version as defined in BIP-174 and BIP-370.
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
+#[repr(u32)]
+#[non_exhaustive]
+pub enum Version {
+    /// PSBT version 0, defined in BIP-174
+    PsbtV0 = 0,
+    /// PSBT version 2, defined in BIP-370
+    PsbtV2 = 2,
+}
+
+impl Version {
+    fn from_standard(version: u32) -> Result<Version, FutureVersionError> {
+        match version {
+            v if v == Version::PsbtV0 as u32 => Ok(Version::PsbtV0),
+            v if v == Version::PsbtV2 as u32 => Ok(Version::PsbtV2),
+            future => Err(FutureVersionError(future))
+
+        }
+    }
+}
+
 /// A Partially Signed Transaction.
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "serde", serde(crate = "actual_serde"))]
 pub struct PartiallySignedTransaction {
     /// The unsigned transaction, scriptSigs and witnesses for each input must be empty.
     pub unsigned_tx: Transaction,
-    /// The version number of this PSBT. If omitted, the version number is 0.
-    pub version: u32,
+    /// The version number of this PSBT. If omitted, the version number defaults to 0.
```

Comment:
Perhaps it should be "if omitted when serialized", since you can't omit it from `enum`. Not sure if people should default to 0 when serializing.

Also not sure if the field should be public. What if a new PSBT feature implies a different version?